cabal-install/Distribution/Client/Init.hs

```diff
@@ -742,6 +742,14 @@ createMainHs flags@InitFlags{ sourceDirs = _
                             , packageType = Flag Executable
                             , mainIs = Flag mainFile } =
   writeMainHs flags mainFile
+createMainHs flags@InitFlags{ sourceDirs = Just (srcPath:_)
```

Comment:
This function has become a bit boilerplatey, you can simplify it by getting rid of pattern matching and adding some helpers.

Reply:
I am not positive if the way I made the change will make things simpler to understand/less boilerplatey or not. Could I get a critique? Thanks!